### ISSUE TITLE
Implement garbage collection or other mechanism to reduce memory bloat in Channel class

### ISSUE THREAD

**Issue by geoknee** [statechannels/go-nitro#106]
For understanding: this is meant to be sparse? A filecoin provider will expect to have a fair number of channels running with lots of micro-transactions in each.

_Originally posted by @NiloCK in https://github.com/statechannels/go-nitro/pull/102#discussion_r767709020_

**Comment by geoknee** [comment]
Whenever we use a durable store, this issue extends beyond "memory bloat" to disk bloat.

We may also be able to save some additional space in the voucher store by e.g. not having the payer store the vouchers.

**Comment by lalexgap** [comment]
I'm going to close this for now as I don't think it's worth optimizing for disk space at this point.

**Comment by geoknee** [comment]
Reopening as it is named under deliverable 1 of the grant https://github.com/filecoin-project/devgrants/issues/1508

**Comment by geoknee** [comment]
A minimal improvement would be to delete channels when they are defunded.

**Comment by geoknee** [comment]
Another easy improvement would be to only store the latest voucher (assuming we don't do so already).

**Comment by geoknee** [comment]
More difficult: delete old states from ledger channels.

**Comment by lalexgap** [comment]
> A minimal improvement would be to delete channels when they are defunded.

We probably still want to keep track of the channel ID of defunded channels, so if we receive a message for a defunded channel we know it's a  stale message and not a new objective.

**Comment by geoknee** [comment]
Yes I think you're right. Somewhat related to https://github.com/statechannels/go-nitro/issues/1116
